sonos/__init__.py

```diff
@@ -2788,119 +2740,283 @@ def _handle_dpt3(self, item, caller=None, source=None, dest=None):
+                elif command == "switch_linein":
+                    sonos_speaker[uid].switch_to_line_in() if item() else None
+                elif command == "switch_tv":
+                    sonos_speaker[uid].switch_to_tv() if item() else None
+                elif command == "play_tunein":
                     start = self._resolve_child_command_bool(item, 'start_after')
                     sonos_speaker[uid].play_tunein(item(), start)
-                if command == "play_url":
+                elif command == "play_url":
                     start = self._resolve_child_command_bool(item, 'start_after')
                     sonos_speaker[uid].play_url(item(), start)
-                if command == "play_sharelink":
+                elif command == "play_sharelink":
                     start = self._resolve_child_command_bool(item, 'start_after')
                     sonos_speaker[uid].play_sharelink(item(), start)
-                if command == "join":
+                elif command == "join":
                     sonos_speaker[uid].join(item())
-                if command == "unjoin":
+                elif command == "unjoin":
                     start = self._resolve_child_command_bool(item, 'start_after')
                     sonos_speaker[uid].unjoin(item(), start)
-                if command == 'load_sonos_playlist':
+                elif command == 'load_sonos_playlist':
                     start = self._resolve_child_command_bool(item, 'start_after')
                     clear_queue = self._resolve_child_command_bool(item, 'clear_queue')
                     track = self._resolve_child_command_int(item, 'start_track')
                     sonos_speaker[uid].load_sonos_playlist(item(), start, clear_queue, track)
-                if command == 'play_tts':
+                elif command == 'play_tts':
                     if item() == "":
+                        self.logger.error("No item value when executing 'play_tts' command")
                         return
                     language = self._resolve_child_command_str(item, 'tts_language', 'de')
                     volume = self._resolve_child_command_int(item, 'tts_volume', -1)
                     fade_in = self._resolve_child_command_bool(item, 'tts_fade_in')
-                    sonos_speaker[uid].play_tts(item(), language, self._local_webservice_path, self._webservice_url,
-                                                volume, self._snippet_duration_offset, fade_in)
-                if command == 'play_snippet':
+                    sonos_speaker[uid].play_tts(item(), language, self._local_webservice_path, self._webservice_url, volume, self._snippet_duration_offset, fade_in)
+                elif command == 'play_snippet' and item() != "":
                     if item() == "":
-                        self.logger.error("No item value when executing play_snippet command")
+                        self.logger.error("No item value when executing 'play_snippet' command")
```

Comment:
This will never be executed. Please doublecheck.

Reply:
Hi, 
should I do the correction within this PR using an additional commit?

Reply:
Yes please

Reply:
Commit with correction is up

Reply:
Great contribution.